File: api/app/routes/health.py
from fastapi import APIRouter, Depends

from app.health_checker import get_cached_results
from app.logger import get_logger
from app.security import require_session

log = get_logger("carver.health")
router = APIRouter(tags=["health"])


@router.get("/")
def root_health():
    # Fallback health route for platforms that probe "/" by default.
    return {"ok": True, "service": "api"}


@router.get("/health")
def health():
    log.debug("Health check")
    return {"ok": True, "service": "api"}


@router.get("/status/services")
def services_status(_session: dict = Depends(require_session)):
    services, last_run = get_cached_results()
    log.info("Status requested | last_run=%s", last_run.isoformat() if last_run else "never")
    return {
        "ok": True,
        "last_run": last_run.isoformat() if last_run else None,
        "next_run_seconds": 300,
        "services": services,
    }

File: jobcarver-go/internal/handlers/health/health.go
// Package health ports api/app/routes/health.py — public liveness probes plus a
// session-gated services-status endpoint.
package health

import (
	"net/http"

	"github.com/go-chi/chi/v5"

	"jobcarver/internal/httpx"
	"jobcarver/internal/security"
)

// Register mounts the health routes onto r.
func Register(r chi.Router) {
	r.Get("/", root)
	r.Get("/health", health)
	r.With(security.RequireSession).Get("/status/services", servicesStatus)
}

func root(w http.ResponseWriter, _ *http.Request) {
	httpx.JSON(w, http.StatusOK, map[string]any{"ok": true, "service": "api"})
}

func health(w http.ResponseWriter, _ *http.Request) {
	httpx.JSON(w, http.StatusOK, map[string]any{"ok": true, "service": "api"})
}

func servicesStatus(w http.ResponseWriter, _ *http.Request) {
	// TODO(go-port): wire to Worker 3's health checker once available. The Python
	// version returns cached background-probe results; until that exists we return
	// the documented empty/unknown fallback.
	httpx.JSON(w, http.StatusOK, map[string]any{
		"ok":               true,
		"last_run":         nil,
		"next_run_seconds": 300,
		"services":         []any{},
	})
}
